src/ecrn_cron_sup.erl

```diff
@@ -28,29 +30,48 @@ start_link() ->
 
 
 %% @doc
-%%  Add a chron job to be supervised
+%% Add a chron job to be supervised
 -spec add_job(erlcron:job_ref(), erlcron:job()) -> erlcron:job_ref().
-add_job(JobRef, Task) ->
-    {ok, _} = supervisor:start_child(?SERVER, [JobRef, Task]),
-    JobRef.
+add_job(JobRef, Job = {_, Task}) when is_function(Task, 2)
+                                    ; is_function(Task, 0)
+                                    ; is_tuple(Task), tuple_size(Task)==3
+                                      , is_atom(element(1,Task))
+                                      , is_atom(element(2,Task))
+                                      , is_list(element(3,Task)) ->
+    case supervisor:start_child(?SERVER, [JobRef, Job]) of
+        {ok, _}                       -> JobRef;
+        {error, {already_started, _}} -> {error, already_started};
+        Other                         -> Other
+    end.
+
+%% @doc
+%% Get a list of all active jobs
+-spec all_jobs() -> [pid()].
+all_jobs() ->
+    [P || {_,P,_,_} <- supervisor:which_children(?SERVER)].
+
+%% @doc
+%% Terminate a job
+terminate(Pid) when is_pid(Pid) ->
+    supervisor:terminate_child(?SERVER, Pid).
 
 %%%===================================================================
 %%% Supervisor callbacks
 %%%===================================================================
 
 %% @private
 init([]) ->
-    RestartStrategy = simple_one_for_one,
-    MaxRestarts = 1000,
-    MaxSecondsBetweenRestarts = 3600,
-
-    SupFlags = {RestartStrategy, MaxRestarts, MaxSecondsBetweenRestarts},
-
-    Restart = transient,
-    Shutdown = 2000,
-    Type = worker,
+    SupFlags = #{
```

Comment:
I'd rather see these maps for the child specs and the supervisor flags being created on a separate private function, just to help the understanding when going cold reading it, but it can stay like this for now.